staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc.go

```diff
@@ -259,17 +224,30 @@ func (p *oidcAuthProvider) idToken() (string, error) {
 		return "", errors.New("No valid id-token, and cannot refresh without refresh-token")
 	}
 
+	p.Refresh()
+
+	return p.cfg[cfgIDToken], nil
+}
+
+func (p *oidcAuthProvider) Refresh() (bool, error) {
```

Comment:
@ericchiang Yes, I didn't want to change the current logic, and I think refresh the token if the token expired is acceptable, do you have any other thoughts ?